src/go/MONIT/annotationManager.go

```diff
@@ -0,0 +1,931 @@
+					log.Fatalf("Unable to marshal the data %+v, error %v\n", rec, err)
+				}
+				if configJSON.Verbose > 0 {
+					log.Printf("Add annotation: %+v", rec)
+				}
+				addAnnotation(data)
+			}
+
+			log.Printf("Annotation Created Successfully\n")
+
+		} else {
+			log.Fatalf("Can't Create annotation. Please provide tags!\n")
+		}
+	} else {
+		log.Fatalf("Can't Create annotation without any text!\n")
+	}
+}
+
+// helper function for updating an annotation which makes PUT request
+func updateAnnotationHelper(annotationID int, data []byte) error {
+
+	var headers [][]string
+	bearer := fmt.Sprintf("Bearer %s", configJSON.Token)
+	h := []string{"Authorization", bearer}
+	headers = append(headers, h)
+	h = []string{"Content-Type", "application/json"}
+	headers = append(headers, h)
+	h = []string{"Accept", "application/json"}
+	headers = append(headers, h)
+
+	apiurl := fmt.Sprintf("%s%s%d", configJSON.GrafanaBaseURL, configJSON.DeleteAnnotationAPI, annotationID)
+	if configJSON.Verbose > 0 {
+		log.Println(apiurl)
+	}
+	req, err := http.NewRequest("PUT", apiurl, bytes.NewBuffer(data))
+	if err != nil {
+		log.Printf("Unable to make request to %s, error: %s", apiurl, err)
+		return err
+	}
+	for _, v := range headers {
+		if len(v) == 2 {
+			req.Header.Add(v[0], v[1])
+		}
+	}
+	if configJSON.Verbose > 1 {
+		dump, err := httputil.DumpRequestOut(req, true)
+		if err == nil {
+			log.Println("request: ", string(dump))
+		}
+	}
+	timeout := time.Duration(configJSON.HTTPTimeout) * time.Second
+	client := &http.Client{Timeout: timeout}
+	resp, err := client.Do(req)
+	if err != nil {
+		log.Printf("Unable to get response from %s, error: %s", apiurl, err)
+		return err
+	}
+	if configJSON.Verbose > 1 {
+		dump, err := httputil.DumpResponse(resp, true)
+		if err == nil {
+			log.Println("response:", string(dump))
+		}
+	}
+
+	if resp.StatusCode != 200 {
+		return fmt.Errorf("HTTP Respose error, code: %d", resp.StatusCode)
+	}
+
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Printf("Unable to read JSON Data from Grafana Annotation PUT API, error: %v\n", err)
+		return err
+	}
+
+	if configJSON.Verbose > 1 {
+		log.Println("response Status:", resp.Status)
+		log.Println("response Headers:", resp.Header)
+		log.Println("response Body:", string(body))
+	}
+
+	return nil
+}
+
+// function which contains the logic of updating an annotation
+func updateAnnotation() {
+	updatedData := make(map[string]interface{})
+
+	if trange != "" {
+		tms := parseTimes(true)
+		if len(tms) == 2 {
+			updatedData["time"] = tms[0]
+			updatedData["timeEnd"] = tms[1]
+		}
+	}
+
+	if annotation != "" {
+		updatedData["text"] = annotation
+	}
+
+	if tags != "" {
+		updatedData["tags"] = parseTags(tags)
+	}
+
+	data, err := json.Marshal(updatedData)
+	if err != nil {
+		log.Fatalf("Unable to parse Data for Updation, update failed !, error: %v", err)
+	}
+
+	err = updateAnnotationHelper(annotationID, data)
+	if err != nil {
```

Comment:
there is no need for error checking here if you'll move this block into updateAnnoataionHelper itself.

Reply:
Fixed